client/allocrunnerv2/alloc_runner.go

```diff
@@ -389,10 +399,18 @@ func getClientStatus(taskStates map[string]*structs.TaskState) (status, descript
 
 // Update the running allocation with a new version received from the server.
 //
-// This method is safe for calling concurrently with Run() and does not modify
-// the passed in allocation.
+// This method sends the updated alloc to Run for serially processing updates.
+// If there is already a pending update it will be discarded and replaced by
+// the latest update.
 func (ar *allocRunner) Update(update *structs.Allocation) {
-	ar.updateCh <- update
+	select {
+	case ar.updateCh <- update:
```

Comment:
Should there be an action for this case? If not maybe add a comment that this is a no-op?